rsconf/component/mongod.py

```diff
@@ -33,6 +33,7 @@ def internal_build_write(self):
         from rsconf.component import logrotate
 
         z = self.j2_ctx.mongod
+        self.append_root_bash_with_main(self.j2_ctx)
```

Comment:
There are two parts to main: installing the rpm and configuring mongod. You should probably split these up. You can append_root_bash_with_resource to install the rpm so that the file installs work and then append with another resource that configures mongod (without the last restart which seems unnecessary).  The configuration needs a sentinel that a user is created, but that probably can be checked via mongosh.

Also the `<<EOF` should be `<<'EOF`` so that bash doesn't accidentally evaluate a password with a dollar sign or whatever.